### PR TITLE
Added splash screen

### PR DESCRIPTION
**Pull request by batchatT**

### CODE REVIEW

File: app/src/main/java/com/example/trempelapp/data_layer/UserInfoRepository.kt [hunk 1]
```diff
@@ -1,11 +1,10 @@
...
-import com.example.trempelapp.data_layer.network.LoginResponse
 import io.reactivex.Single
 
 interface UserInfoRepository {
 
-    fun getLoginStatusByToken()
+    fun getLoginStatusByToken(): Single<String>
 
-    fun loginUser(): Single<LoginResponse>
+//    fun loginUser(): Single<LoginResponse>
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/data_layer/UserInfoRepository.kt:9]: remove it

File: app/src/main/java/com/example/trempelapp/data_layer/UserInfoRepositoryImpl.kt [hunk 2]
```diff
@@ -12,11 +10,11 @@ class UserInfoRepositoryImpl @Inject constructor(
...
-    override fun getLoginStatusByToken() {
-        TODO()
+    override fun getLoginStatusByToken(): Single<String> {
+        return preferences.getUserLoginStatus()
     }
 
-    override fun loginUser(): Single<LoginResponse> {
-        return userService.loginUser(UserCredentials("dsa", "asd"))
-    }
+//    override fun loginUser(): Single<LoginResponse> {
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/data_layer/UserInfoRepositoryImpl.kt:17]: remove it as well

File: app/src/main/java/com/example/trempelapp/data_layer/inMemory/SharedPreferencesManagerImpl.kt [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
+import com.example.trempelapp.utils.TOKEN_PREFERENCES_FILE
+import io.reactivex.Single
+import javax.inject.Inject
 
-    override fun getToken() {
-        TODO("Not yet implemented")
+class SharedPreferencesManagerImpl @Inject constructor(context: Context) :
+    SharedPreferencesManager {
+
+    private val tokenKey = ""
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/data_layer/inMemory/SharedPreferencesManagerImpl.kt:13]: empty string as key??

File: app/src/main/java/com/example/trempelapp/data_layer/inMemory/SharedPreferencesManagerImpl.kt [hunk 4]
```diff
@@ -1,8 +1,23 @@
...
+import android.content.Context
+import android.content.SharedPreferences
+import android.util.Log
+import com.example.trempelapp.utils.TOKEN_PREFERENCES_FILE
+import io.reactivex.Single
+import javax.inject.Inject
 
-    override fun getToken() {
-        TODO("Not yet implemented")
+class SharedPreferencesManagerImpl @Inject constructor(context: Context) :
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/data_layer/inMemory/SharedPreferencesManagerImpl.kt:10]: do not store context here, as you are storing SharedPreferences here

File: app/src/main/java/com/example/trempelapp/presentation_layer/splash_screen/TrempelSplashActivity.kt [hunk 5]
```diff
@@ -20,8 +23,25 @@ class TrempelSplashActivity : BaseActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         (application as TrempelApplication).trempelApp.inject(this)
+        trempelSplashViewModel.injectDagger(application as TrempelApplication)
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentation_layer/splash_screen/TrempelSplashActivity.kt:26]: you inject nothing here. I think it's redundant

File: app/src/main/java/com/example/trempelapp/presentation_layer/splash_screen/TrempelSplashActivity.kt [hunk 6]
```diff
@@ -20,8 +23,25 @@ class TrempelSplashActivity : BaseActivity() {
...
+
         super.onCreate(savedInstanceState)
         supportActionBar?.hide()
         setContentView(binding.root)
+
+        trempelSplashViewModel.checkUserStatus()
```
Comment (denisque75) [inline app/src/main/java/com/example/trempelapp/presentation_layer/splash_screen/TrempelSplashActivity.kt:32]: call it in the setUpViewModel() method

File: app/src/main/java/com/example/trempelapp/data_layer/UserInfoRepository.kt (hunk 1 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/data_layer/UserInfoRepository.kt:9]: done

File: app/src/main/java/com/example/trempelapp/data_layer/UserInfoRepositoryImpl.kt (hunk 2 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/data_layer/UserInfoRepositoryImpl.kt:17]: done

File: app/src/main/java/com/example/trempelapp/data_layer/inMemory/SharedPreferencesManagerImpl.kt (hunk 3 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/data_layer/inMemory/SharedPreferencesManagerImpl.kt:13]: done

File: app/src/main/java/com/example/trempelapp/data_layer/inMemory/SharedPreferencesManagerImpl.kt (hunk 4 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/data_layer/inMemory/SharedPreferencesManagerImpl.kt:10]: done

File: app/src/main/java/com/example/trempelapp/presentation_layer/splash_screen/TrempelSplashActivity.kt (hunk 5 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/splash_screen/TrempelSplashActivity.kt:26]: done

File: app/src/main/java/com/example/trempelapp/presentation_layer/splash_screen/TrempelSplashActivity.kt (hunk 6 above)
Comment (batchatT) [reply app/src/main/java/com/example/trempelapp/presentation_layer/splash_screen/TrempelSplashActivity.kt:37]: done

**Pull request merged**